File: tarea 3/En C++/ejercicio 6.cpp
#include <iostream>
using namespace std;

int main() {
    long long n;

    cout << "Ingrese un numero entero positivo: ";
    cin >> n;

    if (n <= 0) {
        cout << "Debe ingresar un numero positivo." << endl;
        return 0;
    }

    cout << "Secuencia de Collatz:" << endl;

    while (n != 1) {
        cout << n << " ";

        if (n % 2 == 0) {
            n = n / 2;
        } else {
            n = 3 * n + 1;
        }
    }

    cout << 1 << endl;
    cout << "La secuencia llego a 1." << endl;

    return 0;
}

File: tarea 3/En Python/ejercicio_6.py
n = int(input("Ingrese un número entero positivo: "))

if n <= 0:
    print("Debe ingresar un número positivo.")
else:
    print("Secuencia de Collatz:")

    while n != 1:
        print(n, end=" ")

        if n % 2 == 0:
            n = n // 2
        else:
            n = 3 * n + 1

    print(1)
    print("La secuencia llegó a 1.")
    